servicetalk-opentracing-zipkin-publisher/src/main/java/io/servicetalk/opentracing/zipkin/publisher/ZipkinPublisher.java

```diff
@@ -278,6 +149,53 @@ public void onSpanFinished(final InMemorySpan span, long durationMicros) {
             builder.kind(Span.Kind.CLIENT);
         }
 
-        channel.writeAndFlush(builder.build());
+        Span s = builder.build();
+        reporter.report(s);
+    }
+
+    /**
+     * Attempts to flush and close the configured {@link Reporter}.
+     *
+     * @return a {@link Completable} that is completed when the flush and close is done
+     */
+    @Override
+    public Completable closeAsync() {
```

Comment:
We should also override `closeAsyncGracefully()` to correctly delegate the same to the `Reporter`. Typically we create an `AsyncCloseable` in the constructor and delegate to it from respective methods. In this case the `AsyncCloseable` can be created in the constructor like:

```java
closeable = AsyncCloseables.toAsyncCloseable(graceful -> {
    Completable flush = Completable.completed();
    if (reporter instanceof Flushable) {
        flush = globalExecutionContext().executor().submit(() -> {
            try {
                ((Flushable) reporter).flush();
            } catch (IOException e) {
                logger.error("Exception while flushing reporter: {}", e.getMessage(), e);
            }
        });
    }
    // ST Reporters can implement AsyncCloseable so we wouldn't have to call a blocking close in that case
    Completable close = Completable.completed();
    if (reporter instanceof AsyncCloseable) {
        close = graceful ? ((AsyncCloseable) reporter).closeAsyncGracefully() :
                ((AsyncCloseable) reporter).closeAsync();
    } else if (reporter instanceof Closeable) {
        close = globalExecutionContext().executor().submit(() -> {
            try {
                ((Closeable) reporter).close();
            } catch (IOException e) {
                logger.error("Exception while closing reporter: {}", e.getMessage(), e);
            }
        });
    }
    return flush.concat(close);
});
```